### PR TITLE
Improve AES/XTS performance for 6x unrolling with vpermxor instruction for PowerPC

### PR DESCRIPTION
**Pull request by dannytsen**
Improve AES/XTS performance of 6x unrolling by using vpermxor instruction instead.  It combines xor and vperm in one instruction.   It passed all the tests and with improved speed test performance up to 19% for larger message size.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by t8m** [comment]
This is not eligible for CLA: trivial. Could you please sign a regular CLA?

**Comment by t8m** [comment]
Is the vpermxor instruction available on all CPUs the assembler code is targeted for?

**Comment by dannytsen** [comment]
> This is not eligible for CLA: trivial. Could you please sign a regular CLA and remove the CLA: trivial annotation from the commit message?

Sure.  I'll sign one.

**Comment by dannytsen** [comment]
> 

Yes.  vpermxor is available in P8 and abobe, ISA 2.07, 3.0 and 3.1.

**Comment by dannytsen** [comment]
@t8m I am not sure if your legal received CCLA yet but I have submitted my ICLA.

**Comment by dannytsen** [comment]
@t8m Hi, Is this PR ready for merge?  Or still waiting for more approval?  Thanks.

**Comment by paulidale** [comment]
It needs a second approval from @openssl/committers

**Review by t8m — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the master branch. Thank you for your contribution.

**Pull request closed without merging**
